### PR TITLE
Extend verify config output with compatible target types

### LINKED ISSUE #2096: Extend `verify-config` with some info useful in IDEs
#2095 fixes IDE errors when using `copy_target` feature without `target` set in the config. However, the user can still pick invalid targets in the popup. `mirrord verify-config` should return a list of available target types, so that the popup is not confusing.

### PR DESCRIPTION
**Pull request by Razz4780**
Closes #2096

### CODE REVIEW

File: mirrord/cli/src/verify_config.rs [hunk 1]
```diff
@@ -56,6 +57,30 @@ impl From<TargetConfig> for VerifiedTargetConfig {
...
+
+impl TargetType {
+    fn all() -> impl Iterator<Item = Self> {
+        [Self::Targetless, Self::Pod, Self::Deployment, Self::Rollout].into_iter()
+    }
+
+    fn compatible_with(&self, config: &FeatureConfig) -> bool {
+        match self {
+            Self::Targetless | Self::Rollout => !config.copy_target.enabled,
+            Self::Pod => !(config.copy_target.enabled && config.copy_target.scale_down),
```
Comment (meowjesty) [inline mirrord/cli/src/verify_config.rs:78]: Do we have these config matches documented anywhere (don't remember for the `copy_target` stuff)?

If we don't could you doc it saying why they're not allowed.

File: changelog.d/2096.added.md [hunk 2]
```diff
@@ -0,0 +1 @@
+`mirrord verify-config` not outputs a list of available target types.
```
Comment (meowjesty) [inline changelog.d/2096.added.md:1]:
```suggestion
`mirrord verify-config` now outputs a list of available target types.
```

typo?

**Review by meowjesty — APPROVED** [review]
:+1: 

Approved as I only really have doc comments.

File: changelog.d/2096.added.md (hunk 2 above)
Comment (Razz4780) [reply changelog.d/2096.added.md:1]: typo typo, thanks

File: mirrord/cli/src/verify_config.rs (hunk 1 above)
Comment (Razz4780) [reply mirrord/cli/src/verify_config.rs:78]: Updated `copy_target` docs, now it mentions compatibility with target types :+1:

**Pull request merged**